test/keysym.c

```diff
@@ -182,6 +204,33 @@ main(void)
     assert(test_utf8(0x110ffff, "\xf4\x8f\xbf\xbf"));
     assert(test_utf8(0x1110000, NULL) == 0);
 
+    assert(test_utf32_to_keysym('y', XKB_KEY_y));
+    assert(test_utf32_to_keysym('u', XKB_KEY_u));
+    assert(test_utf32_to_keysym('m', XKB_KEY_m));
+    assert(test_utf32_to_keysym(0x43c, XKB_KEY_Cyrillic_em));
+    assert(test_utf32_to_keysym(0x443, XKB_KEY_Cyrillic_u));
+    assert(test_utf32_to_keysym('!', XKB_KEY_exclam));
+    assert(test_utf32_to_keysym(0xF8, XKB_KEY_oslash));
+    assert(test_utf32_to_keysym(0x5D0, XKB_KEY_hebrew_aleph));
+    assert(test_utf32_to_keysym(0x634, XKB_KEY_Arabic_sheen));
+
+    assert(test_utf32_to_keysym('\b', XKB_KEY_BackSpace));
+    assert(test_utf32_to_keysym('\t', XKB_KEY_Tab));
+    assert(test_utf32_to_keysym('\n', XKB_KEY_Linefeed));
+    assert(test_utf32_to_keysym(0x0b, XKB_KEY_Clear));
+    assert(test_utf32_to_keysym('\r', XKB_KEY_Return));
+    assert(test_utf32_to_keysym(0x1b, XKB_KEY_Escape));
+    assert(test_utf32_to_keysym(0x7f, XKB_KEY_Delete));
+
+    assert(test_utf32_to_keysym(' ', XKB_KEY_space));
+    assert(test_utf32_to_keysym(',', XKB_KEY_comma));
+    assert(test_utf32_to_keysym('.', XKB_KEY_period));
+    assert(test_utf32_to_keysym('=', XKB_KEY_equal));
+    assert(test_utf32_to_keysym('9', XKB_KEY_9));
+    assert(test_utf32_to_keysym('*', XKB_KEY_asterisk));
+    assert(test_utf32_to_keysym(0xd7, XKB_KEY_multiply));
+    assert(test_utf32_to_keysym('-', XKB_KEY_minus));
```

Comment:
Can you check one that would be directly encoded? Like some emoji or such.

Reply:
I do not want to use direct encoding as character literals behavior in such case can be platform-dependent. A few of already added tests (i.e. hebrew, cyrilic and arabic) uses explicit numeric values instead to avoid such problems.